### ISSUE TITLE
Proposal and Poll: GUI Implementation Fund Match

### ISSUE THREAD

**Issue by jring-o** [gridcoin-community/Gridcoin-Research#1401]
<h1>Final Results</h1>
I can't believe I never posted a final results for the bounty. My mistake.

Anyway, everything should be updated. The community raised a total of 262,000 GRC from 19-21 people _and_ passed and validated the foundation match poll!

The total bounty for building the new GUI is 512,000 GRC.

Anyone interested in working to claim the bounty should first reach out to one of the following core developers:

@denravonska 
@jamescowens 

A request to collect the pledged bounties into a wallet held by the core team is entirely reasonable, though the 250,000 GRC from the foundation is already reserved and ready to go.

<h2>UPDATE 3</h2>
https://steemit.com/gridcoin/@jringo/update-3-3-days-left-to-vote-gridcoin-gui-implementation-fund-match

<h2>UPDATE 2</h2>
https://steemit.com/gridcoin/@jringo/update-2-proposal-and-poll-gridcoin-gui-implementation-fund-match

<h2>UPDATE 1</h2>
The core development team is considering that rebuilding the GUI (not the core) from the ground up in a modern framework might be the best path. This changes nothing with the poll and the bounty. Anyone considering putting work into implementing the GUI is encouraged to start a discussion with the dev team.

---

<h1>Useful Links</h1>

[Assets Proposal](https://steemit.com/gridcoin/@jringo/proposal-and-poll-funding-a-proposed-gridcoin-core-client-ui-redesign) 

[Completion of Assets Poll Thread](https://steemit.com/gridcoin/@jringo/completed-proposal-and-poll-funding-a-proposed-gridcoin-core-client-ui-redesign) 

[GUI Improvements Discussion Thread](https://github.com/gridcoin-community/Gridcoin-Research/issues/847) 
WARNING: Very image heavy thread.

[Completed Assets](https://github.com/madmaxpayne/Gridcoin-wallet-redesign) 

<h1>Overview</h1>

[Reddit Thread](https://www.reddit.com/r/gridcoin/comments/aq7glf/proposal_and_poll_gui_implementation_fund_match/)

[Steemit Thread](https://steemit.com/gridcoin/@jringo/proposal-and-poll-gridcoin-gui-implementation-fund-match)

About 6 months ago the community agreed to spend some time and money funding the design of a new UX and the assets for a subsequent GUI. Those assets are completed and ready for implementation. Madmaxpayne has been paid for their work. As detailed in the links above, there are still funds owed to them once the implementation of the assets is complete; Max agreed to stick around and help anyone implementing the assets as minor tweaks to assets are a common occurrence in this type of thing.

We have been gathering a bounty for the task of building this new GUI from the assets provided. The intent of this bounty is to find someone with the necessary skills who also wants to invest themselves in the future development of the Gridcoin protocol, network, and community. To achieve this goal we have marked the bounty in GRC _only_. So far we have the following commitments from community members:

Startail - 50,000 GRC
Vortac - 30,000 GRC
Jim - 25,000 GRC
Gunde - 25,000 GRC
sodom89 - 27,000 GRC
Dc7d - 25,000 GRC
An0n - 15,000 GRC
ShmoogleOsukami - 10,000 GRC
H202 - 10,000 GRC
philipswift - 10,000 GRC
dblanch - 10,000 GRC
ilikechocolate - 7,000
Barton26 - 5,000 GRC
Dennis - 5,000 GRC
EnzoCaricoTri - 2,500 (Held by jringo - 10fc8a069e762ebf9d33e71979eb2744a6f297ebdb076452d32e86485f80e21b)
Anonymous Donations - 2,500 GRC (Held by jringo) 
Jringo - 1,000 GRC
Hunkymonk -1,000 GRC
Nethlek - 1,000 GRC

Total - 262,000 GRC

These individuals will hold their own bounties until they are satisfied with the GUI implementation.

If you would like to pledge your own GRC to this task post the amount pledged on the Github thread. If you do not have a Github account and still wish to pledge reach out to jringo on discord or on slack and he will update the thread on Github.

You can continue to pledge to the bounty until the bounty is claimed.

<h1>PROPOSAL</h1>

This proposal requests a funds match from the foundation up to 250,000 GRC. These funds would be reserved and held by the foundation until the new GUI is implemented. The new GUI should use the UX and assets designed by Madmaxpayne. The completed assets are linked at the top of this document and directly below. Madmaxpayne can be reached on slack.

[Completed Assets](https://github.com/madmaxpayne/Gridcoin-wallet-redesign)

<h2>How The Fund Match Works</h2>

This fund match would operate much like other fund matches. There are three scenarios:

1. Under 250,000 GRC is pledged by the community
2. Exactly 250,000 GRC is pledged by the community
3. Over 250,000 GRC is pledged by the community

1 - Under: If 50,000 GRC is the total pledged by the community, for example, the foundation would pledge another 50,000 GRC. This would bring the total bounty to 100,000 GRC.

2 - Exact: If 250,000 GRC is the total pledged by the community, the foundation would pledge another 250,000 GRC bringing the total bounty to 500,000 GRC.

3 - Over: If 300,000 GRC is the total pledged by the community, for example, the foundation would pledge another 250,000 GRC bringing the total bounty to 550,000 GRC.

The amount pledged by the foundation will match the bounty at the time the bounty is claimed. No more than 250,000 GRC will be pledged by the foundation if this proposal is approved.

<h2>Requirements for Claiming Bounty</h2>

- Complete and Functional GUI using the assets provided by Madmaxpayne
- Completion signed off by jringo, ravon, barton26, madmaxpayne, NeuralMiner, and Jim Owens.

<h2>Questions? Comments? Concerns?</h2>

The Github thread is the main discussion thread as it is saved nice and cozy next to other development discussions.

However if you have any questions, comments, or concerns you can post them on any thread and I will do my best to cross-post major content. You can also join us on slack or discord and someone will be sure to have an answer or someone will connect you with someone who does. If you want to join slack (where most development takes place) ask for an invite.

[Discord](https://discord.me/gridcoin)

<h1>Poll</h1>

Duration: 2 weeks [14 Days]

Start Time: February 13th
End Time: February 27th 15:09 UTC

Question: 
Do you approve of the proposed fund match for the bounty for implementing the new GUI?

Answers:
Yes
No
Abstain

Validation:

60% or more AVW participation
As we are still experimenting with and observing the behavior of AVW, a TVW of 15% will also validate the poll

**Comment by grc-H202** [comment]
I'll pledge 10,000 GRC!

**Comment by sodom89** [comment]
I'll pledge 20,000 GRC!

**Comment by philipswift** [comment]
I'll pledge 10,000 GRC

**Comment by jring-o** [comment]
The core development team is considering that rebuilding the GUI (not the core) from the ground up in a modern framework might be the best path. This changes nothing with the poll and the bounty. Anyone considering putting work into implementing the GUI is encouraged to start a discussion with the dev team.

**Comment by Vortac2** [comment]
I pledge 30,000 GRC

**Comment by HunkyMonk** [comment]
I'll pledge 1,000 GRC

**Comment by startailcoon** [comment]
I'll pledge 50,000 GRC for the help of implementing this new UX.

**Comment by dblanch256** [comment]
I pledge 10,000 GRC.

**Comment by sodom89** [comment]
I will increase my pledge to 27,000 GRC so we have full 250,000GRC

**Comment by dblanch256** [comment]
Bravo, sodom89!

**Comment by philipswift** [comment]
Hurrah! superb team work :) beer o'clock then !

**Comment by EnzoCaricoTri** [comment]
I pledge 2500 GRC

**Comment by sibebleuze** [comment]
Hi, I think this is a great idea. However I'm here to point out that the poll in GridcoinResearch doesn't link to this discussion and neither to the steemit thread. Instead it links to https://www.reddit.com/r/gridcoin/comments/84xtij/a_tour_of_the_gridcoin_wallet/, which is a nice post explaining the app, but not very useful if you're looking for info on the poll itself. Can anyone here change the link so it links to the appropriate posts? I only just found this because there's a link in the next poll and I can imagine other people to have missed this discussion as well.

Secondly, shouldn't there be more than one option in a poll? Because now the one option has 100%, which isn't very representative of how the opinions really are. Someone who doesn't approve (I imagine they do in fact exist), doesn't have the option to vote against the proposal, which doesn't seem very fair to me.

**Comment by jamescowens** [comment]
@sibebleuze I don't think you are looking at the right poll. This poll... https://www.gridcoinstats.eu/poll/proposal:_gui_implementation_fund_match does point here. There are a couple of invalid polls put up that are confusing.

**Comment by presciencia** [comment]
I have just upgraded to latest Ingrid release and the new GUI looks really good! I am wondering is @cyrossignol qualified for this bounty now? As years passed and this is still open. There is much work of him on this GUI as in linked PR's from #847 for example

- https://github.com/gridcoin-community/Gridcoin-Research/pull/1739
- https://github.com/gridcoin-community/Gridcoin-Research/pull/1819
- https://github.com/gridcoin-community/Gridcoin-Research/pull/2096
- https://github.com/gridcoin-community/Gridcoin-Research/pull/2102
- https://github.com/gridcoin-community/Gridcoin-Research/pull/2106
- https://github.com/gridcoin-community/Gridcoin-Research/pull/2114
- https://github.com/gridcoin-community/Gridcoin-Research/pull/2115
- https://github.com/gridcoin-community/Gridcoin-Research/pull/2117
- https://github.com/gridcoin-community/Gridcoin-Research/pull/2126
- https://github.com/gridcoin-community/Gridcoin-Research/pull/2138
- https://github.com/gridcoin-community/Gridcoin-Research/pull/2143
- https://github.com/gridcoin-community/Gridcoin-Research/pull/2145
- https://github.com/gridcoin-community/Gridcoin-Research/pull/2151
- https://github.com/gridcoin-community/Gridcoin-Research/pull/2169
- https://github.com/gridcoin-community/Gridcoin-Research/pull/2170

**Comment by jamescowens** [comment]
No… He intended this specifically as a stopgap measure until the bounty could be satisfied.

**Comment by presciencia** [comment]
Ok I am not aware, sorry.

What is remaining to be satisfied in the bounty?

**Comment by cyrossignol** [comment]
@presciencia Thanks for the compliment. As I wrote in #2102, I waived a claim to this bounty for the work that you linked above. I implemented the designs using our current GUI framework, and we'd prefer that someone address the bounty with a more modern tool set. I chose not to rebuild every screen, and there are some finer details in the proposed design that I left for the final job. We also discussed the implementation of a client/server GUI as a desirable feature.

**Comment by presciencia** [comment]
Ok thanks for the explanations @cyrossignol. I have these questions to ask now.

What is the benefit you are looking for a modern tool set? QT is a great modern tool for c++ project, is there better tools? Is full rewrite effort worth because the design is may be 90% done by your last work? What is the reason to not finish with current tools?

As I was interested to do or help with this rewrite but it looked to need years of work, so I am asking because you did this very fast with current tools. The people above who pledge may be gone after years or even now, it is possible that a person doing the full rewrite can not claim the full bounty (just foundation) and decided not to work. The pledge is collected with a trusted person?

I am not sure there is ever the full rewrite with these conditions. It is your choice to waive payment, but it is the most needed progress I have seen in this coin and you finish can be paid for hard work with out waiting again years for another person.

**Comment by cyrossignol** [comment]
@presciencia There are a couple of other folks with early concepts created with Qt Quick/QML (a preferred tool kit for the rewrite). I do not wish to step on their toes. I used our current Qt Widgets framework for these changes. I can finish the design with Widgets, but some of the proposed GUI features are more difficult to implement with the older technology, and I'd rather spend time in other areas. Qt Quick may give us better footing for new features in the future. You can reach these people in Slack to ask about a timeline.

I don't know how the pledges were collected. It looks like @jring-o holds some of them.
